cognite/client/data_classes/data_modeling/views.py

```diff
@@ -863,3 +866,72 @@ def dump(self, camel_case: bool = True) -> dict[str, Any]:
             output["connection_type"] = "multi_reverse_direct_relation"
 
         return output
+
+
+@dataclass
+class PropertyUnitReference:
+    property: str
+    unit: UnitReference | UnitSystemReference
+
+    def dump(self, camel_case: bool = True) -> dict[str, Any]:
+        return {"property": self.property, "unit": self.unit.dump(camel_case)}
+
+    @classmethod
+    def load(cls, data: dict) -> PropertyUnitReference:
+        return cls(
+            property=data["property"],
+            unit=UnitReference.load(data["unit"])
+            if "externalId" in data["unit"]
+            else UnitSystemReference.load(data["unit"]),
+        )
+
+
+@dataclass(frozen=True)
+class SourceDef:
```

Comment:
Looks like we already have `SourceSelector`. let's just use that instead.. Will need to introduce a new superclass for that though since `properties` shouldn't be supplied when used in context of `list`.

so smth like
```
@dataclass
class SourceSelectorWithoutProperties(CogniteObject):
   source: ViewId
   target_units: list[PropertyUnit]

@dataclass
class SourceSelector(SourceSelectorWithoutProperties):
   properties: list[str]
```